test/fakeintake/client/client.go

```diff
@@ -81,9 +81,21 @@ const (
 // ErrNoFlareAvailable is returned when no flare is available
 var ErrNoFlareAvailable = errors.New("no flare available")
 
+// Option is a configuration option for the client
+type Option func(*Client)
+
+// WithStrictFakeintakeIDCheck enables strict fakeintake ID check
+func WithStrictFakeintakeIDCheck() Option {
+	return func(c *Client) {
+		c.strictFakeintakeIDCheck = true
+	}
+}
```

Comment:
I wonder if we shouldn't do the opposite: enable it by default and have an option to disable it. What do you think?

Reply:
Ideally we want to use it everywhere yes, but I did not want to change the default behavior to not potentially break existing things

Reply:
Things are already broken if the fakeintake restarted. What exactly do you think this new feature would break? I think this change would be pretty harmless

Reply:
Sounds good, updated the default behavior